plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -912,12 +912,27 @@ public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHan
     @Override
     public void createSchema(ConnectorSession session, String schemaName)
     {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            if (connection.getMetaData().storesUpperCaseIdentifiers()) {
+                schemaName = schemaName.toUpperCase(ENGLISH);
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
         execute(session, "CREATE SCHEMA " + quoted(schemaName));
     }
 
     @Override
     public void dropSchema(ConnectorSession session, String schemaName)
     {
+        JdbcIdentity identity = JdbcIdentity.from(session);
+        try (Connection connection = connectionFactory.openConnection(session)) {
```

Comment:
Since we're opening connection anyway, let's move `execute` below here, and reuse the connection

same above

Reply:
Done!